### DISCUSSION TITLE
Question about my specific example, using filters [Solved!]

### DISCUSSION THREAD

**Discussion by ErandM50** [bertrandmartel/tableau-scraping#40] [Q&A] [answered]
Hi when i try on this link [https://public.tableau.com/app/profile/roy.morgan.research/viz/PublicCovid-19MovementDataMel_A/Visitorpage](url) it is giving me the error below:

`from tableauscraper import TableauScraper as TS

url = 'https://public.tableau.com/app/profile/roy.morgan.research/viz/PublicCovid-19MovementDataMel_A/Visitorpage'

ts = TS()
ts.loads(url)
dashboard = ts.getDashboard()
Traceback (most recent call last):

  File "<ipython-input-82-acfc514a1fb6>", line 6, in <module>
    ts.loads(url)

  File "C:\ProgramData\Anaconda4\lib\site-packages\tableauscraper\TableauScraper.py", line 80, in loads
    soup.find("textarea", {"id": "tsConfigContainer"}).text

AttributeError: 'NoneType' object has no attribute 'text'`

Can you tell me what the issue might be?

**Comment by bertrandmartel** [comment, marked as answer]
@ErandM50 The tableau url is: https://public.tableau.com/views/PublicCovid-19MovementDataMel_A/Visitorpage. You can locate this url from the network tab of Chrome dev console. It seems it calls the base path `/app/profile` but then uses `/views` url to get the data:

```python
from tableauscraper import TableauScraper as TS

url = "https://public.tableau.com/views/PublicCovid-19MovementDataMel_A/Visitorpage"

ts = TS()
ts.loads(url)
workbook = ts.getWorkbook()

for t in workbook.worksheets:
    print(f"worksheet name : {t.name}")  # show worksheet name
    print(t.data)  # show dataframe for this worksheet

```

**Reply by ErandM50** [reply]
Thank you for your reply @bertrandmartel!
This worked for me. I was wondering if there is a way within the library to get the data for all dropdowns (cities, places of interest) or a way I can modify this to do it myself. Right now it is getting the data for the default graph that appears when you first go to the page (Melbourne - Melbourne CBD).

**Reply by bertrandmartel** [reply]
@ErandM50 Those are filters. Filters are bound to a worksheet. You can get all the filters like this for the specific sheet:

```python
print(workbook.getWorksheetNames())
print(workbook.getWorksheet("POI Unique Devices.v").getFilters())
```
Checkout: https://github.com/bertrandmartel/tableau-scraping#set-filter

**Comment by ErandM50** [comment]
@bertrandmartel amazing! This works perfectly for me. I appreciate your work on this and the support!
